code/game/objects/items/weapons/grenades/wirebomb.dm

```diff
@@ -0,0 +1,286 @@
+	SSobj.processing -= src
+	return ..()
+
+/obj/item/weapon/grenade/wirebomb/proc/detonate()
+	if(!armed || detonating)
+		return
+	detonating = 1
+	for(var/V in spikes)
+		var/obj/item/wirebomb_spike/spike = V
+		explosion(get_turf(spike), 0, 1, 2, 3)
+	explosion(get_turf(src), 0, 1, 2, 3)
+
+/obj/item/weapon/grenade/wirebomb/attack_hand(mob/user)
+	if(armed)
+		detonate()
+	else if(!anchored)
+		..()
+
+/obj/item/weapon/grenade/wirebomb/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/wirecutters))
+		if(!armed)
+			return
+		user << "<span class='warning'>You start disarming [src]. If you stop now it will go off!</span>"
+		if(do_after(user, 400, 20, 1, src))
+			disarm()
+		else
+			detonate()
+	else if(armed)
+		detonate()
+	else
+		..()
+
+/obj/item/weapon/grenade/wirebomb/proc/disarm()
+	armed = 0
+	for(var/V in spikes)
+		var/obj/item/wirebomb_spike/spike = V
+		spike.owner = null
+		spike.icon_state = "bombspike_disarmed"
+		if(spike.wire)
+			spike.wire.End()
+			spike.wire = null
+		spikes -= spike
+	anchored = 0
+	icon_state = "wirebomb_disarmed"
+
+/obj/item/weapon/grenade/wirebomb/singularity_pull(S, current_size)
+	if(current_size >= STAGE_FIVE)
+		detonate()
+
+
+////////////////////
+//Wired spike bomb//
+////////////////////
+/obj/item/wirebomb_spike
+	name = "bomb spike"
+	desc = "It explodes when triggered by its tripwire"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "bombspike"
+	throwforce = 20
+	var/obj/item/weapon/grenade/wirebomb/owner
+	var/atom/sticked_to
+	var/datum/beam/wire
+
+/obj/item/wirebomb_spike/proc/launch(direction, range, passflag=0)
+	var/count = 0
+	dir = direction
+	while(count < range)
+		count++
+		var/atom/A = SimpleMove(direction, passflag)
+		if(A)
+			sticked_to = A
+			if(istype(A, /atom/movable))
+				var/atom/movable/AM = A
+				glue_object(AM)
+				if(AM.anchored)
+					anchored=1
+				if(ismob(AM))
+					var/mob/M = AM
+					M.hitby(src,1,0)
+			else
+				anchored = 1
+			layer = A.layer+0.1
+			break
+		sleep(1)
+
+	owner.flying_spike_count--
+	if(!sticked_to)
+		qdel(src)
+
+/obj/item/wirebomb_spike/ex_act()
+	if(owner)
+		if(owner.armed && !owner.detonating)
+			owner.detonate()
+			return
+	else
+		explosion(get_turf(src), 0, 1, 2, 3)
+	qdel(src)
+
+/obj/item/wirebomb_spike/singularity_pull(S, current_size)
+	if(current_size >= STAGE_FIVE)
+		if(owner)
+			owner.detonate()
+
+/obj/item/wirebomb_spike/attack_hand(mob/user)
+	if(!sticked_to)
+		anchored = 0
+	if(!anchored && !glued_objects.len)
+		..()
+	else
+		user << "<span class='warning'>It's wedged into [sticked_to]. You can't pick it up.</span>"
+
+/obj/item/wirebomb_spike/glued_move(atom/movable/AM)
+	..()
+	if(owner)
+		owner.detonate()
+
+/obj/item/wirebomb_spike/unglue_object(atom/movable/AM)
+	..()
+	if(owner)
+		owner.detonate()
+
+/obj/item/wirebomb_spike/Move()
+	if(..())
+		if(owner)
+			owner.detonate()
+
+/obj/item/wirebomb_spike/proc/SimpleMove(direction, passflag=0)
```

Comment:
Because current Move produces bad results for what I want here, like not moving diagonally.